src/Common/Site_Health/Factory.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace TEC\Common\Site_Health;
+
+use Tribe__Main;
+
+/**
+ * Class Factory
+ *
+ * @link https://developer.wordpress.org/reference/hooks/debug_information/
+ *
+ * @since   TBD
+ *
+ * @package TEC\Common\Site_Health
+ */
+class Factory {
+	/**
+	 * Generates a Generic field from a set of arguments.
+	 *
+	 * @since TBD
+	 *
+	 * @param string      $id
+	 * @param string      $label
+	 * @param string|null $value
+	 * @param int         $priority (optional) By default all fields are generated with priority 50.
+	 *
+	 * @return Info_Field_Abstract
+	 */
+	public function generate_generic_field( string $id, string $label, ?string $value, int $priority = 50 ): Info_Field_Abstract {
+		return Fields\Generic_Info_Field::from_args( $id, $label, $value, $priority );
```

Comment:
Just curious - why do we need a factory? Couldn't it just be 
```suggestion
		return new Fields\Generic_Info_Field( $id, $label, $value, $priority );
```
If it's not necessary, I'd learn toward the above to avoid complexity/confusion, but I could be missing something.

Reply:
@bordoni this piece is some of the code I inherited from you. I tend to agree with @stratease - the from_args and the constructor are nearly identical, is this DRY?

Thinking on it - the static method can be called without instantiating the object, but that seems a bit silly here?